chatterbot/adapters/adaptation.py

```diff
@@ -0,0 +1,35 @@
+from chatterbot.utils.module_loading import import_module
+
+
+class Adaptation(object):
+    """
+    This is a base from which ChatterBot inherits utility methods
+    and a context attribute that allows access of adapters to be
+    shared between other adapters. This also makes it possible to
+    share other context information such as a name, or the current
+    conversation with each of the adapters.
+    """
+
+    class context(object):
```

Comment:
Actually, there might be a problem with this. I had thought that static classes in python could be updated with out effecting each other _if_ they were inside on an instance of another class. So for example...

```python
class A:
    class B:
        x = 0

# then ...

a = A()
b = A()

a.B.x = 1
b.B.x = 2

print a.B.x
>>> 1
print b.B.x
>>> 2
```

However this does not appear to be the case. It looks like the subclass (B) is actually updated across every instance of A. Do you know if it would be possible to keep class B from changing other instances of A?